File: 문장속단어.java

import java.util.*;

class Main {
  public String solution(String str) {
      String answer = "";
      int m = Integer.MIN_VALUE;
      String[] s = str.split(" ");

      for(String t: s) {
          int len = t.length();
          if(len > m) {
              m = len;
              answer = t;
          }
      }


      return answer;
  }
  
  public static void main(String[] args){
    Main T = new Main();
    
Scanner in=new Scanner(System.in);
  String str = in.nextLine();
    System.out.println(T.solution(str));
    
  }
}


import java.util.*;

class Main {
  public String solution(String str) {
      String answer = "";
      int m = Integer.MIN_VALUE, pos;
      
      while((pos=str.indexOf(' ')) != -1) {
          String tmp = str.substring(0, pos);
          int len = tmp.length();

          if( len > m ) {
              len = m;
              answer = tmp;
          }

          str = str.substring(pos+1);
      }
      if(str.length() > m) answer = str;
      
      return answer;
  }
  
  public static void main(String[] args){
    Main T = new Main();
    
Scanner in=new Scanner(System.in);
  String str = in.nextLine();
    System.out.println(T.solution(str));
    
  }
}

File: 문장속단어.kt
import java.util.*

class Main{
    fun solution(str: String) : String {
        var answer = "";
        var m = Int.MIN_VALUE;
        val s = str.split(" ").toTypedArray();

        for(t in s) {
            val len = t.length
            if(len > m) {
                m = len
                answer = t
            }
        }

        return t
    }

    fun main(args: Array<String>) {
        val T = Main();
        val scanner = Scanner(System.in)
        val str = scanner.nextLine()
        println(T.solution(str));
    }
}

class Main{
    fun solution(str: String) : String {
        var answer = ""
        var m = Int.MIN_VALUE
        var pos = 0

        while((pos=str.split(" ").indexOf()) != -1) {
            var t = str.substring(0, pos)
            var len = t.length()
            if(len > m) {
                m = len
                answer = t
            }

            str = str.substring(pos+1)
        }

        if(str.length > m) answer = str

        return t
    }

    fun main(args: Array<String>) {
        val T = Main();
        val scanner = Scanner(System.in)
        val str = scanner.nextLine()
        println(T.solution(str))
    }
}